lua/lint/linters/pylint.lua

```diff
@@ -16,19 +16,22 @@ return {
     local diagnostics = {}
     local buffer_path = vim.fn.fnamemodify(vim.api.nvim_buf_get_name(bufnr), ":~:.")
 
+    output = string.gsub(output, "message%-id", "message_id")
     for _, item in ipairs(vim.fn.json_decode(output) or {}) do
       if not item.path or vim.fn.fnamemodify(item.path, ":~:.") == buffer_path then
         local column = 0
         if item.column > 0 then
           column = item.column - 1
         end
         table.insert(diagnostics, {
+          source = 'pylint',
           lnum = item.line - 1,
           col = column,
           end_lnum = item.line - 1,
           end_col = column,
           severity = assert(severities[item.type], 'missing mapping for severity ' .. item.type),
           message = item.message,
+          code = item.message_id,
```

Comment:
```suggestion
    for _, item in ipairs(vim.fn.json_decode(output) or {}) do
      if not item.path or vim.fn.fnamemodify(item.path, ":~:.") == buffer_path then
        local column = 0
        if item.column > 0 then
          column = item.column - 1
        end
        table.insert(diagnostics, {
          source = 'pylint',
          lnum = item.line - 1,
          col = column,
          end_lnum = item.line - 1,
          end_col = column,
          severity = assert(severities[item.type], 'missing mapping for severity ' .. item.type),
          message = item.message,
          user_data = {
            lsp = {
              code = item['message-id'],
            },
          },
```

No need to replace the message id column in the string. table access is possible for words with hyphens as well.
And there is no first class `code` property (see `:help diagnostic-structure`). For LSP items it's put into the `user_data` property.

Reply:
Oh, I have just seen the mapping you do in `from_pattern`. Somehow it was working. Fixed done! I also had to update the tests.